dali/pipeline/operators/tensorRTInfer/tensorRTInfer.h

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_TENSORRTINFER_TENSORRTINFER_H_
+#define DALI_PIPELINE_OPERATORS_TENSORRTINFER_TENSORRTINFER_H_
+
+#include <cuda_runtime_api.h>
+#include <common.h>
+#include <dlfcn.h>
+#include <utility>
+#include <vector>
+#include <map>
+#include <string>
+#include <NvInfer.h>
+#include <NvInferPlugin.h>
+#include "dali/common.h"
+#include "dali/error_handling.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/pipeline/operators/operator.h"
+
+namespace dali {
+
+typedef struct {
+    // Binding Index
+    int binding_index;
+    // DataType for each bindings
+    DALIDataType data_type;
+    // Shape required for allocating memory to output tensorlist
+    std::vector<Dims> dali_dimensions;
+}BindingParam;
+
+template <typename Backend> class TensorRTInfer : public Operator<Backend> {
+ public:
+  explicit inline TensorRTInfer(const OpSpec &spec) :
+    Operator<Backend>(spec),
+    input_blobs_(spec.GetRepeatedArgument<std::string>("input_blobs")),
+    output_blobs_(spec.GetRepeatedArgument<std::string>("output_blobs")),
+    plugins_(spec.GetRepeatedArgument<std::string>("plugins")),
+    trt_batch_size_(spec.GetArgument<int64>("trt_batch_size")),
+    num_outputs_(spec.GetArgument<int>("num_outputs")),
+    engine_data_(spec.GetArgument<std::string>("engine")),
+    use_dla_core_(spec.GetArgument<int>("use_dla_core")) {
+    for (auto &s : plugins_) {
+        void *dlh = dlopen(s.c_str(), RTLD_LAZY);
+        if (nullptr == dlh) {
+          std::cout << "Error while performing dlopen on the" << s << " library"
+                    << dlerror() << std::endl;
+        }
+     }
+    initLibNvInferPlugins(&trt_logger_, "");
+    runtime_ = nvinfer1::createInferRuntime(trt_logger_);
+    if (-1 != use_dla_core_ && use_dla_core_ < MAX_ALLOWED_DLA_CORE) {
+        runtime_->setDLACore(use_dla_core_);
+    }
+    engine_ = runtime_->deserializeCudaEngine(engine_data_.data(), engine_data_.size(), nullptr);
+    context_ = engine_->createExecutionContext();
+
+    DALI_ENFORCE(use_dla_core_ < MAX_ALLOWED_DLA_CORE,
```

Comment:
Bellow enforce should be put at the beginning before we start to load plugins and do other stuff.